### PR TITLE
fix isequal_normalized for combining-char reordering

### LINKED ISSUE #52408: `isequal_normalized("בְּ", Unicode.normalize("בְּ")) == false`
`Unicode.isequal_normalized(x, Unicode.normalize(x))` should probably return true in all cases.

Here's an example where it doesn't
```julia
julia> using Unicode

julia> str = String([0xd6, 0xbc, 0xd6, 0xb0])
"ְּ"

julia> Unicode.isequal_normalized(str, Unicode.normalize(str))
false
```

Thanks to Neel Smith on Zulip for finding this

### PR DESCRIPTION
**Pull request by stevengj**
Fixes #52408.

(Note that this function was added in Julia 1.8, in #42493.)

In the future it would be good to further optimize this function by adding a fast path for the common case of strings that are mostly ASCII characters.  Perhaps simply skip ahead to the first byte that doesn't match before we begin doing decomposition etcetera.

### CODE REVIEW

File: stdlib/Unicode/src/Unicode.jl [hunk 1]
```diff
@@ -251,29 +261,79 @@ julia> isequal_normalized(s1, "NOËL", casefold=true)
...
-        return 1, n, iterate(s, state)
+
+        # sort by combining class
+        if n < 32 # almost always true
+            for j1 = 2:n # insertion sort
+                cc = combining_class(d[j1])
+                iszero(cc) && continue # don't re-order non-combiners
+                for j2 = j1:-1:2
+                    cc1 = combining_class(d[j2-1])
+                    cc1 ≤ cc && break
```
Comment (stevengj) [inline stdlib/Unicode/src/Unicode.jl:307]: slight simplification:
```suggestion
                    combining_class(d[j2-1]) ≤ cc && break
```

**Comment by stevengj** [comment]
CI failures seem unrelated.

**Comment by stevengj** [comment]
Should be good to merge?

**Review by StefanKarpinski — COMMENTED** [review]
LGTM (already merged)

**Pull request merged**